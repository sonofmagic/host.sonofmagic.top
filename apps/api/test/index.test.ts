import { sql } from 'drizzle-orm'

// String.raw
function css(template: {
  raw: readonly string[] | ArrayLike<string>
}, ...substitutions: any[]) {
  console.log(template, substitutions)
}

describe('sql', () => {
  it('should work', () => {
    expect(sql`1 ${'12'}`).toMatchSnapshot()
  })

  it('css', () => {
    const selector = '.apap'
    const color = 'rgba(1,1,1,0.1)'
    expect(css`
      .a{
        color: red;
      }

      ${selector}{
        --xx-yy: ${color};
        color: var(--xx-yy);
      }
      `).toMatchSnapshot()
  })
})
