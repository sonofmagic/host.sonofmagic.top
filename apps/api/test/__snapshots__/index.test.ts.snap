// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sql > css 1`] = `undefined`;

exports[`sql > should work 1`] = `
SQL {
  "decoder": {
    "mapFromDriverValue": [Function],
  },
  "queryChunks": [
    StringChunk {
      "value": [
        "1 ",
      ],
    },
    "12",
    StringChunk {
      "value": [
        "",
      ],
    },
  ],
  "shouldInlineParams": false,
  "usedTables": [],
}
`;
